migrateBuckets/migrateBuckets.js

```diff
@@ -0,0 +1,327 @@
+        log.getSerializedUids(), err => {
+            if (err) {
+                log.error(`Error updating object ${Key} metadata: ${err}`);
+                return cb(err);
+            }
+            log.debug(`Object ${Key} id successfully updated`);
+            return cb();
+        }, putOptions);
+    }, getOptions);
+}
+
+function _changeObjectsAccount(bucket, log, cb) {
+    const bucketName = bucket.bucketName;
+    const versioningEnabled = bucket.versioningEnabled;
+    const listMethod = versioningEnabled ? 'listObjectVersions' : 'listObjects';
+    const dataKey = versioningEnabled ? 'Versions' : 'Contents';
+    let moreObjects = true;
+    let objectsArr = [];
+    let marker = null;
+    async.whilst(
+        () => moreObjects === true,
+        done => {
+            const listParams = { Bucket: bucketName };
+            if (marker) {
+                listParams.Marker = marker;
+            }
+            s3[listMethod](listParams, (err, data) => {
+                if (err) {
+                    return done(err);
+                }
+                objectsArr = objectsArr.concat(data[dataKey]);
+                if (data.IsTruncated) {
+                    marker = data.Contents[data[dataKey].length - 1].Key;
+                } else {
+                    moreObjects = false;
+                }
+                return done();
+            });
+        },
+        err => {
+            if (err) {
+                log.error('Error listing objects from bucket ' +
+                    `${bucketName}: ${err}`);
+                return cb(err);
+            }
+            return async.each(objectsArr,
+                (object, done) =>
+                    _changeOneObjectAccount(bucket, object, log, done),
+                err => {
+                    if (err) {
+                        return cb(err);
+                    }
+                    log.debug(`All objects in bucket ${bucketName} ` +
+                        'successfully updated');
+                    return cb();
+                }
+            );
+        }
+    );
+}
+
+function migrateBuckets(bucket, cb) {
+    async.waterfall([
+        next => _changeBucketAccount(bucket, logger, next),
+        (versioningEnabled, next) => {
+            /* eslint-disable no-param-reassign */
+            bucket.versioningEnabled = versioningEnabled;
+            _changeObjectsAccount(bucket, logger, next);
+        },
+    ], err => {
+        if (err) {
+            logger.error(`Bucket ${bucketName} failed to process: ${err}`);
+            return cb(err);
+        }
+        logger.info(`Bucket ${bucketName} transfer is complete`);
+        return _changeUsersBucket(bucket, logger, cb);
+    });
+}
+
+migrateBuckets(commander, err => {
+    if (err) {
+        logger.error(`Bucket migration failed: ${err}\n` +
+            'Please run script again');
+    } else {
+        logger.info('Completed bucket migration');
+    }
+});
```

Comment:
It could be good to add signal handlers, in case an admin stops the script with `SIGINT/SIGTERM` (see other scripts in s3utils for examples).